resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/JettyClientEngine.java

```diff
@@ -0,0 +1,223 @@
+package org.jboss.resteasy.client.jaxrs.engines;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.client.InvocationCallback;
+import javax.ws.rs.client.ResponseProcessingException;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.client.util.DeferredContentProvider;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.util.Callback;
+import org.jboss.resteasy.client.jaxrs.AsyncClientHttpEngine;
+import org.jboss.resteasy.client.jaxrs.internal.ClientInvocation;
+import org.jboss.resteasy.client.jaxrs.internal.ClientResponse;
+
+public class JettyClientEngine implements AsyncClientHttpEngine {
+    private static final InvocationCallback<ClientResponse> NOP = new InvocationCallback<ClientResponse>() {
+        @Override
+        public void completed(ClientResponse response) {
+        }
+
+        @Override
+        public void failed(Throwable throwable) {
+        }
+    };
+
+    private final HttpClient client;
+
+    public JettyClientEngine(HttpClient client) {
+        if (!client.isStarted()) {
+            try {
+                client.start();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```

Comment:
Is this how this library prefers to handle the construction of these engines?  Seems unfortunate to depend on using an object that throw a checked exception, but then not have handling for checked exceptions for the construction of these engines.